tests/src/main/java/com/linkedin/kmf/tests/BasicPipelineTest.java

```diff
@@ -0,0 +1,293 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.tests;
+
+import com.linkedin.kmf.services.ConsumeService;
+import com.linkedin.kmf.services.DefaultMetricsReporterService;
+import com.linkedin.kmf.services.JettyService;
+import com.linkedin.kmf.services.JolokiaService;
+import com.linkedin.kmf.services.ProduceService;
+import com.linkedin.kmf.services.TopicManagementService;
+import com.linkedin.kmf.services.configs.CommonServiceConfig;
+import com.linkedin.kmf.services.configs.ConsumeServiceConfig;
+import com.linkedin.kmf.services.configs.DefaultMetricsReporterServiceConfig;
+import com.linkedin.kmf.services.configs.ProduceServiceConfig;
+import com.linkedin.kmf.services.configs.TopicManagementServiceConfig;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.apache.kafka.common.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static net.sourceforge.argparse4j.impl.Arguments.store;
+
+/*
+ * The BasicPipelineTest test is intended to monitor the health and performance of kafka brokers across a mirror maker
+ * pipeline. It creates one producer and one consumer with the given configuration, produces messages with increasing
+ * integer in the message String, consumes messages, and keeps track of number of lost messages, duplicate messages, e2e
+ * latency, throughput, etc. As well, BasicPipeLineTest creates a topic management service for every kafka and zookeeper
+ * broker in the pipeline.
+ *
+ * BasicPipelineTest test exports these metrics via JMX. It also periodically report metrics if DEBUG level logging
+ * is enabled. This information can be used by other application to trigger alert when kafka brokers fail. It also
+ * allows users to track pipeline performance, e.g. latency and throughput.
+ */
+
+public class BasicPipelineTest implements Test {
```

Comment:
The first thing we need to think about BasicPipelineTest is its interface, i.e. how do you want users to specify its configuration. Note that there are two different ways to specify configuration, i.e. command line and config file. Let's work on the config file first since the command line approach will depend on the config file approach. In your current patch, you have asked user to specify `consume.props` for consume service and similarly `produce.props` for produce  service. An alternative is to ask user to provide bootstrap.server and zookeeper.url of source cluster and destination cluster directly, maybe with config name such as source.zookeeper.url which will be specific to BasicPipelineTest, and construct properties map for ProduceService and ConsumeService based on the user-provided config.

I personally think the 2nd approach is better because user doesn't have to know how to config ProduceService and ConsumeService when configuring BasicPipelineTest, 2) it can potentially make the config more intuitive with dedicated config name (e.g. source.zookeeper.url), and 3) less config level. It means that we need to have an config class for each test, similar to what we already have for each service. We can discuss this design when we meet next time.

Also, suppose you go with current approach of having consume.props, I don't think it should be in the ConsumeServiceConfig. For example, let's say in the future we have a test which needs to create to ConsumeService for both source and destination cluster. How are you going to specify the config for this test? Instead of having a config with name `consume.props`, you probably need `source.consume.props` and `destination.consume.props`. It means that this config name should be specific to each test instead of being part of the service.